Remotion/Validation/UnitTests/Rules/ObjectValidationRuleTest.cs

```diff
@@ -1,21 +1,193 @@
+      validatorStub
+          .Setup(_ => _.Validate(It.IsAny<ObjectValidatorContext>()))
+          .Returns(
+              new[]
+              {
+                new ObjectValidationFailure(
+                    15,
+                    "15 is not a number",
+                    "Localized validation message: 15 is not a number.")
+              });
+
+      var rule = new ObjectValidationRule<object>(null, new[] { validatorStub.Object });
+
+      var result = rule.Validate(context);
+
+      Assert.That(result, Is.Empty);
+    }
+
+    [Test]
+    public void Validate_WithUnfulfilledCondition_ReturnsEmptyResultAsValidatorsAreNotCalled ()
+    {
+      Func<object, bool> condition = _ => false;
+      var context = new ValidationContext(new object());
+
+      var validatorStub = new Mock<IObjectValidator>();
+      validatorStub
+          .Setup(_ => _.Validate(It.IsAny<ObjectValidatorContext>()))
+          .Returns(
+              new[]
+              {
+                new ObjectValidationFailure(
+                    15,
+                    "15 is not a number",
+                    "Localized validation message: 15 is not a number.")
+              });
+
+      var rule = new ObjectValidationRule<object>(condition, new[] { validatorStub.Object });
+
+      var result = rule.Validate(context);
+
+      Assert.That(result, Is.Empty);
     }
 
     [Test]
-    [Ignore("TODO RM-5906")]
-    public void IsActive ()
+    public void IsActive_WithoutInstanceToValidate_ReturnsFalse ()
     {
-      // consider better name than "IsActive"
+      Func<object, bool> condition = _ => true;
+
+      var context = new ValidationContext(null);
+      var rule = new ObjectValidationRule<object>(condition, new List<IObjectValidator>());
+
+      var result = rule.IsActive(context);
+
+      Assert.That(result, Is.False);
+    }
+
+    [Test]
+    public void IsActive_WithNoCondition_ReturnsTrue ()
+    {
+      var context = new ValidationContext(new object());
+      var rule = new ObjectValidationRule<object>(null, new List<IObjectValidator>());
+
+      var result = rule.IsActive(context);
+
+      Assert.That(result, Is.True);
+    }
+
+    [Test]
+    [TestCase(true)]
+    [TestCase(false)]
+    public void IsActive_ReturnsValueFromCondition (bool expectedResult)
+    {
+      Func<object, bool> condtition = _ => expectedResult;
+
+      var context = new ValidationContext(new object());
+      var rule = new ObjectValidationRule<object>(condtition, new List<IObjectValidator>());
+
+      var result = rule.IsActive(context);
+
+      Assert.That(result, Is.EqualTo(expectedResult));
+    }
+
+    [Test]
+    public void ToString_WithObjectType_ReturnsFullyQualifiedName ()
+    {
+      const string expectedResult = "ObjectValidationRule (System.Object)";
+      var rule = new ObjectValidationRule<object>(null, new List<IObjectValidator>());
+
+      var result = rule.ToString();
+
+      Assert.That(result, Is.EqualTo(expectedResult));
```

Comment:
Stylistic things:
When there's no special cases, we don't need the "WithSomething" phrase in the test title. Here' it would just be "ToString_ReturnsValidatorTypeAndFullNameOfValidatedType". The existing option is good enough, though.

In case of the expected result, unless there's sepecial logic involved, it's inlined at the assertion. Usually, the more simple the situation, the more is inlined. That way, you see it when you look at the Assert instead of needing to hunt down the right variable inside the test.

Using System.Object instead of somehting test-specifc like Person from the test domain makes the test less safe. In particular, System.Object, you could easily use that one accidentally in production code and never notice it via a red test. That's why I try to use either obsucre .NET BCL types or types from the actual test assembly, whenever I can.